roachtest: add index backfill perturbation test

andrewbaptist opened the pull request:
This commit adds three new tests
perturbation/(metamorphic|full|dev)/backfill which will perform an index
backfill and measure the latency impact.

During the startup of the test, it creates a second database named
"backfill" and fills it with 10K values. During the perturbation phase
it creates a large index on the backfill database by indexing all the
keys and values.

Fixes: https://github.com/cockroachdb/cockroach/issues/128392

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130047)
<!-- Reviewable:end -->

Commit:
roachtest: add index backfill perturbation test

This commit adds three new tests
perturbation/(metamorphic|full|dev)/backfill which will perform an index
backfill and measure the latency impact.

During the startup of the test, it creates a second database named
"backfill" and fills it with 10K values. During the perturbation phase
it creates a large index on the backfill database by indexing all the
keys and values.

Fixes: #128392

Release note: None

kvoli reviewed (commented):
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130047)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @andrewbaptist, @renatolabs, and @vidit-bhat)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6MD6eLCTspPpW3PW9h):*
Add the full name of the variation added in the commit:

`perturbation/(metamorphic|full|dev)/backfill`
___
*[`pkg/cmd/roachtest/tests/admission_control_latency.go` line 332 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6MDvLD9spMnAJiNAxh:-O6MDvLD9spMnAJiNAxi:b-7bj1xv) ([raw file](https://github.com/cockroachdb/cockroach/blob/785ff4078d09371ad76e24491d00c3bb5ad16dfa/pkg/cmd/roachtest/tests/admission_control_latency.go#L332)):*
> ```Go
> 	defer db.Close()
> 
> 	cmd := fmt.Sprintf("ALTER DATABASE backfill CONFIGURE ZONE USING constraints = '[+node%d]', lease_preferences='[[-node%d]]'", target, target)
> ```

Does this need `num_replicas` as well?
___
*[`pkg/cmd/roachtest/tests/admission_control_latency.go` line 343 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6ME4mWBF7-TCjn7wVy:-O6ME4mWBF7-TCjn7wVz:b-xerttw) ([raw file](https://github.com/cockroachdb/cockroach/blob/785ff4078d09371ad76e24491d00c3bb5ad16dfa/pkg/cmd/roachtest/tests/admission_control_latency.go#L343)):*
> ```Go
> 	// to create a lot of SSTables and ranges in a short amount of time.
> 	runCmd := fmt.Sprintf(
> 		"./cockroach workload run kv --db backfill --duration=%s --max-block-bytes=%d --concurrency=100 {pgurl%s}",
> ```

Should this also have `min_block_bytes`, otherwise the fill will be uniformly random in 1-10k bytes


<!-- Sent from Reviewable.io -->

andrewbaptist reviewed (commented):
TFTR

I cleaned up the commit add added the min-block-bytes.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-:-O6WR0ESDUM-msGyIFne:b7nfcqt)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @kvoli, @renatolabs, and @vidit-bhat)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6MD6eLCTspPpW3PW9h):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

Add the full name of the variation added in the commit:

`perturbation/(metamorphic|full|dev)/backfill`
</blockquote></details>

Done. I wrote more about the test in the commit also.
___
*[`pkg/cmd/roachtest/tests/admission_control_latency.go` line 332 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6MDvLD9spMnAJiNAxh:-O6WQV0c0acpouHJUePX:b-npa398) ([raw file](https://github.com/cockroachdb/cockroach/blob/785ff4078d09371ad76e24491d00c3bb5ad16dfa/pkg/cmd/roachtest/tests/admission_control_latency.go#L332)):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

Does this need `num_replicas` as well?
</blockquote></details>

No it inherits the "default" `num_replicas` as 3. In the future if we wanted to test other replica counts it is probably better to leave this as the default.
___
*[`pkg/cmd/roachtest/tests/admission_control_latency.go` line 343 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6ME4mWBF7-TCjn7wVy:-O6WQxSu5oYB_2M5y95-:b-qy7sib) ([raw file](https://github.com/cockroachdb/cockroach/blob/785ff4078d09371ad76e24491d00c3bb5ad16dfa/pkg/cmd/roachtest/tests/admission_control_latency.go#L343)):*
<details><summary><i>Previously, kvoli (Austen) wrote…</i></summary><blockquote>

Should this also have `min_block_bytes`, otherwise the fill will be uniformly random in 1-10k bytes
</blockquote></details>

Added. 


<!-- Sent from Reviewable.io -->

kvoli reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-:-O6h7cbw-YcjmZId8BWm:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andrewbaptist, @renatolabs, and @vidit-bhat)

___
*[`pkg/cmd/roachtest/tests/admission_control_latency.go` line 332 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130047#-O6MDvLD9spMnAJiNAxh:-O6h7_DD72s7YE5dglLi:bb7ohn6) ([raw file](https://github.com/cockroachdb/cockroach/blob/785ff4078d09371ad76e24491d00c3bb5ad16dfa/pkg/cmd/roachtest/tests/admission_control_latency.go#L332)):*
<details><summary><i>Previously, andrewbaptist (Andrew Baptist) wrote…</i></summary><blockquote>

No it inherits the "default" `num_replicas` as 3. In the future if we wanted to test other replica counts it is probably better to leave this as the default.
</blockquote></details>

sgtm


<!-- Sent from Reviewable.io -->

andrewbaptist commented:
TFTR! I'm looking forward to seeing this running by Monday!

bors r=kvoli

andrewbaptist commented:
bors retry
